src/dialog-controller.ts

```diff
@@ -97,17 +97,20 @@ export class DialogController {
       return this.closePromise;
     }
 
-    return this.closePromise = invokeLifecycle(this.controller.viewModel || {}, 'canDeactivate').catch(reason => {
-      this.closePromise = undefined;
-      return Promise.reject(reason);
+    const dialogResult = { wasCancelled: !ok, output } as DialogCloseResult;
+
+    return this.closePromise = invokeLifecycle(this.controller.viewModel || {}, 'canDeactivate', dialogResult)
+      .catch(reason => {
+        this.closePromise = undefined;
+        return Promise.reject(reason);
```

Comment:
Stupid question, but how do you expect the indentation to be ? :blush: 
Reason for splitup is the 200 chars limitation.